fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/transactionprocessor/impl/AdvancedPaymentScheduleTransactionProcessor.java

```diff
@@ -496,7 +532,15 @@ private void processSingleTransaction(LoanTransaction loanTransaction, MonetaryC
              * For existing transactions, check if the re-payment breakup (principal, interest, fees, penalties) has
              * changed.<br>
              */
-            final LoanTransaction newLoanTransaction = LoanTransaction.copyTransactionProperties(loanTransaction);
+            LoanTransaction newLoanTransaction = LoanTransaction.copyTransactionProperties(loanTransaction);
+            LocalDate transactionDate = loanTransaction.getTransactionDate();
+            if (loanTransaction.getTypeOf().equals(REAMORTIZE)) {
```

Comment:
I dont think we need this.
I think the original logic should collect the outstanding principal amounts and check whether the new balance is matching with the original transaction amount or not.

Reply:
fixed